compiler/rustc_ast/src/attr/mod.rs

```diff
@@ -233,14 +233,20 @@ impl AttrItem {
 
 impl Attribute {
     /// Returns `true` if it is a sugared doc comment (`///` or `//!` for example).
-    /// So `#[doc = "doc"]` will return `false`.
+    /// So `#[doc = "doc"]` (which is a doc comment) and `#[doc(...)]` (which is not
```

Comment:
`#[doc = "doc"]` is not a doc comment, and not a comment at all, it's a doc attribute.
I'm not sure why this causes confusion, the function name tells exactly what it does.

Reply:
Because `///` is a doc comment and syntactic sugar over `#[doc = "..."]`, so the confusion is still quite strong.